restartmanager/restartmanager.go

```diff
@@ -72,8 +75,12 @@ func (rm *restartManager) ShouldRestart(exitCode uint32, hasBeenManuallyStopped
 	}
 	if rm.timeout == 0 {
 		rm.timeout = defaultTimeout
-	} else {
+	} else if rm.timeout > maxRestartTimeout {
+		return false, nil, fmt.Errorf("reached max restart timeout")
+	} else if rm.timeout < maxBaseTimeout {
 		rm.timeout *= backoffMultiplier
+	} else {
+		rm.timeout *= time.Duration(rand.Intn(9) + 1) // Increase timeout by random offset to prevent a situation where all containers are trying to restart simultaneously after a reboot
```

Comment:
Actually I'd remove this and `maxBaseTimeout`, keep on using the `backoffMultiplier` as previously stated, the `maxRestartTimeout` fixes the issue already.

As for the randomness `containerd` doesn't let us start more than 10 containers at a time atm to avoid overloading the system